### DISCUSSION TITLE
[error] Command '/opt/so/__salt.tmp.ze0j_sot' failed with return code: 1

### DISCUSSION THREAD

**Discussion by foxman0719** [Security-Onion-Solutions/securityonion#12501] [2.4]
### Version

2.4.50

### Installation Method

Security Onion ISO image

### Description

installation

### Installation Type

Standalone

### Location

on-prem with Internet access

### Hardware Specs

Exceeds minimum requirements

### CPU

16

### RAM

40

### Storage for /

314GB

### Storage for /nsm

1664GB

### Network Traffic Collection

span port

### Network Traffic Speeds

1Gbps to 10Gbps

### Status

Yes, all services on all nodes are running OK

### Salt Status

Yes, there are salt failures (please provide detail below)

### Logs

Yes, there are additional clues in /opt/so/log/ (please provide detail below)

### Detail

good afternoon dear community, today I installed the tool for the first time, I have many doubts but to start at the end the following error appeared, but when I see in the GRID everything is OK.

<<<<ERROR>>>>

### Cleaning up temp files in /nsm/elastic-agent-workspace[INFO    ] Creating module dir '/var/cache/salt/minion/extmods/grains'
[INFO    ] Syncing grains for environment 'base'
[INFO    ] Loading cache from salt://_grains, for base
[INFO    ] Caching directory '_grains/' for environment 'base'
[INFO    ] Loading fresh modules for state activity
[INFO    ] Fetching file from saltenv 'base', ** done ** 'elasticfleet/install_agent_grid.sls'
[INFO    ] Executing command /bin/systemctl in directory '/root'
[INFO    ] Running state [salt://elasticfleet/files/so_agent-installers/so-elastic-agent_linux_amd64] at time 22:06:25.984017
[INFO    ] Executing state cmd.script for [salt://elasticfleet/files/so_agent-installers/so-elastic-agent_linux_amd64]
[INFO    ] Fetching file from saltenv 'base', ** done ** 'elasticfleet/files/so_agent-installers/so-elastic-agent_linux_amd64'
[INFO    ] Executing command '/opt/so/__salt.tmp.zy71yos4' in directory '/opt/so'
[ERROR   ] Command '/opt/so/__salt.tmp.zy71yos4' failed with return code: 1
[ERROR   ] stdout: 
Installation initiated, view install log for further details.
[ERROR   ] stderr: 

Installing in non-interactive mode.: signal: killed
[ERROR   ] retcode: 1
[ERROR   ] {'pid': 251613, 'retcode': 1, 'stdout': '\nInstallation initiated, view install log for further details.', 'stderr': '\n\nInstalling in non-interactive mode.: signal: killed'}
[INFO    ] Completed state [salt://elasticfleet/files/so_agent-installers/so-elastic-agent_linux_amd64] at time 22:09:43.364017 (duration_in_ms=197380.001)
[WARNING ] State is set to retry, but a valid dict for retry configuration was not found.  Using retry defaults
[INFO    ] State result does not match retry until value, state will be re-run in 30 seconds
[INFO    ] Running state [salt://elasticfleet/files/so_agent-installers/so-elastic-agent_linux_amd64] at time 22:10:13.396974
[INFO    ] Executing state cmd.script for [salt://elasticfleet/files/so_agent-installers/so-elastic-agent_linux_amd64]
[INFO    ] Executing command '/opt/so/__salt.tmp.nzmcebme' in directory '/opt/so'
[ERROR   ] Command '/opt/so/__salt.tmp.nzmcebme' failed with return code: 1
[ERROR   ] stdout: 
Installation initiated, view install log for further details.
[ERROR   ] stderr: 

Installing in non-interactive mode.: signal: killed
[ERROR   ] retcode: 1
[ERROR   ] {'pid': 259298, 'retcode': 1, 'stdout': '\nInstallation initiated, view install log for further details.', 'stderr': '\n\nInstalling in non-interactive mode.: signal: killed'}
[INFO    ] Completed state [salt://elasticfleet/files/so_agent-installers/so-elastic-agent_linux_amd64] at time 22:13:22.115889 (duration_in_ms=188718.915)
local:
----------
          ID: run_installer
    Function: cmd.script
        Name: salt://elasticfleet/files/so_agent-installers/so-elastic-agent_linux_amd64
      Result: False
     Comment: Attempt 1: Returned a result of "False", with the following comment: "Command 'salt://elasticfleet/files/so_agent-installers/so-elastic-agent_linux_amd64' run"
              Command 'salt://elasticfleet/files/so_agent-installers/so-elastic-agent_linux_amd64' run
     Started: 22:06:25.984016
    Duration: 416098.91599999997 ms
     Changes:   
              ----------
              pid:
                  259298
              retcode:
                  1
              stderr:
                  
                  
                  Installing in non-interactive mode.: signal: killed
              stdout:
                  
                  Installation initiated, view install log for further details.

Summary for local
------------
Succeeded: 0 (changed=1)
Failed:    1
------------
Total states run:     1
Total run time: 416.099 s

thank you for you help!

### Guidelines

- [X] I have read the discussion guidelines at https://github.com/Security-Onion-Solutions/securityonion/discussions/1720 and assert that I have followed the guidelines.

**Comment by reyesj2** [comment]
What is the output of ```sudo elastic-agent status```

**Reply by foxman0719** [reply]
![image](https://github.com/Security-Onion-Solutions/securityonion/assets/68893069/ee306984-71c3-4322-8874-a44d036d8b1c)

nothing appears

**Comment by foxman0719** [comment]
Buenos días, sigo teniendo el mismo error durante la instalación ya he reinstalado en varias ocasiones, quien me colabora por favor

**Reply by reyesj2** [reply]
It looks like the elastic agent wasn't able to install. 
Look at the log /opt/so/SO-Elastic-Agent_Installer.log (Share the log if you can)

Then you can try to rerun the Elastic Agent installer by running (Review the log again if it fails a second time)
``` sudo salt-call state.apply elasticfleet.install_agent_grid```

**Comment by foxman0719** [comment]
output of: /opt/so/SO-Elastic-Agent_Installer.log

 timestamp=2024-03-19T18:39:56.588557266Z level=info message="Version Information" ElasticAgentVersion=8.7.0 WrapperVersion=2.4.2
timestamp=2024-03-19T18:39:56.58864759Z level=info message="Runtime Data" EnrollmentToken="ajQ3a1Y0NEI3LVZQa2FUZFN1YTA6eGlsQ1U1d0pUNkNRSG9yc200MlpoUQ==" FleetURL/s=https://172.24.252.117:8220,https://emcicadmin:8220
timestamp=2024-03-19T18:39:56.58867157Z level=info message="Installation Progress" Status="Starting Installation Precheck"
timestamp=2024-03-19T18:39:56.614637997Z level=info message="Installation Progress" FleetHostConnectivityCheck=Success FleetHostURL=https://172.24.252.117:8220
timestamp=2024-03-19T18:39:56.614760841Z level=info message="Installation Progress" Status="Fleet Host is accessible - Continuing installation"
timestamp=2024-03-19T18:39:56.614790567Z level=info message="Installation Progress" Status="Installation Precheck Complete"
timestamp=2024-03-19T18:39:56.614823469Z level=info message="Installation Progress" Status="Extracting Elastic Agent files"
timestamp=2024-03-19T18:40:03.304755427Z level=info message="Installation Progress" Status="Executing Elastic Agent installer"
timestamp=2024-03-19T18:40:03.3048229Z level=info message="Installation Progress" Status="Executing the following: ./so-elastic-agent_source/elastic-agent/elastic-agent install --url=https://172.24.252.117:8220 --enrollment-token=ajQ3a1Y0NEI3LVZQa2FUZFN1YTA6eGlsQ1U1d0pUNkNRSG9yc200MlpoUQ== --certificate-authorities=/opt/Elastic/SO/soca.crt -n"
timestamp=2024-03-19T18:43:03.404846418Z level=error message="Installation Progress" Context="Installing in non-interactive mode." ErrorDetails="signal: killed"
timestamp=2024-03-19T18:43:35.648958214Z level=info message="Version Information" ElasticAgentVersion=8.7.0 WrapperVersion=2.4.2
timestamp=2024-03-19T18:43:35.649188383Z level=info message="Runtime Data" EnrollmentToken="ajQ3a1Y0NEI3LVZQa2FUZFN1YTA6eGlsQ1U1d0pUNkNRSG9yc200MlpoUQ==" FleetURL/s=https://172.24.252.117:8220,https://emcicadmin:8220
timestamp=2024-03-19T18:43:35.649256866Z level=info message="Installation Progress" Status="Starting Installation Precheck"
timestamp=2024-03-19T18:43:35.676539177Z level=info message="Installation Progress" FleetHostConnectivityCheck=Success FleetHostURL=https://172.24.252.117:8220
timestamp=2024-03-19T18:43:35.676607333Z level=info message="Installation Progress" Status="Fleet Host is accessible - Continuing installation"
timestamp=2024-03-19T18:43:35.676626719Z level=info message="Installation Progress" Status="Installation Precheck Complete"
timestamp=2024-03-19T18:43:35.676644847Z level=info message="Installation Progress" Status="Extracting Elastic Agent files"
timestamp=2024-03-19T18:43:42.132912646Z level=info message="Installation Progress" Status="Executing Elastic Agent installer"
timestamp=2024-03-19T18:43:42.133000825Z level=info message="Installation Progress" Status="Executing the following: ./so-elastic-agent_source/elastic-agent/elastic-agent install --url=https://172.24.252.117:8220 --enrollment-token=ajQ3a1Y0NEI3LVZQa2FUZFN1YTA6eGlsQ1U1d0pUNkNRSG9yc200MlpoUQ== --certificate-authorities=/opt/Elastic/SO/soca.crt -n"
timestamp=2024-03-19T18:46:42.139293459Z level=error message="Installation Progress" Context="Installing in non-interactive mode." ErrorDetails="signal: killed"
